.github/workflows/connector-dri-pr-reviewer.yml

```diff
@@ -0,0 +1,118 @@
+        uses: tj-actions/changed-files@v44
+        with:
+          files: docs/reference/connectors/**
+
+      - name: Assign DRI based on changed connector docs
+        if: steps.changed_connector_files.outputs.all_modified_files != ''
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{secrets.DOCS_GITHUB_TOKEN}}
+          script: |
+            // Helper function to fetch DRI information by property and value
+            async function getDRIByProperty(property, value) {
+              const url = `https://website-api.hasura.io/docs-services/docs-server/dri/lookup?property=${encodeURIComponent(property)}&value=${encodeURIComponent(value)}`;
+              
+              try {
+                const response = await fetch(url);
+                
+                if (!response.ok) {
+                  console.error(`Error fetching DRI data: ${response.status} ${response.statusText}`);
+                  return null;
+                }
+                
+                const data = await response.json();
+                return data.length > 0 ? data[0] : null;
+              } catch (error) {
+                console.error(`Failed to fetch DRI data: ${error.message}`);
+                return null;
+              }
+            }
+
+            const changedFiles = `${{ steps.changed_connector_files.outputs.all_modified_files }}`.split(' ').filter(f => f.length > 0);
+            if (changedFiles.length === 0) {
+              console.log("No relevant connector documentation files changed.");
+              return;
+            }
+            console.log("Changed connector files:", changedFiles);
+
+            const assignees = new Set();
+            const connectorPathRegex = /^docs\/reference\/connectors\/([^\/]+)\//;
+
+            for (const filePath of changedFiles) {
+              const match = filePath.match(connectorPathRegex);
+              
+              if (match && match[1]) {
+                const connectorFolder = match[1];
+                console.log(`Looking up DRI for connector folder: ${connectorFolder}`);
+                
+                // Fetch DRI information from the API
+                const connectorInfo = await getDRIByProperty('registry-folder', connectorFolder);
+                
+                if (connectorInfo && connectorInfo['dri-github-handle'] && connectorInfo['dri-github-handle'].toLowerCase() !== 'null') {
+                  let handle = connectorInfo['dri-github-handle'];
+                  if (handle.startsWith('@')) {
+                    handle = handle.substring(1);
+                  }
+                  if (handle) {
+                    assignees.add(handle);
+                    console.log(`Found DRI ${handle} for connector ${connectorFolder}`);
+                  }
+                } else {
+                  console.log(`No DRI found for connector folder: ${connectorFolder}`);
+                }
+              }
+            }
```

Comment:
To make this a bit more efficient, could we:
- Move the call to the API out of the loop
- Create a set of all the unique folders which have changes and then make the API call

I'm thinking through what may happen if there's (likely) multiple files within a single directory which have changes...if I'm reading the current implementation correctly, this would create redundant calls.